Add PHP7 and MongoDB package support

connorcook opened the pull request:
Adds a small package tweak to the work Jose Luis Falcon did so it can be merged back into the original repository without breaking packages.

jlfalcon committed:
Increased PHP version

jlfalcon committed:
Update composer.json

jlfalcon committed:
Update composer.json

jlfalcon committed:
Update composer.json

jlfalcon committed:
Update composer.json

Commit:
Adapted to ext-mongodb

Commit:
Updated README

Commit:
Fix bug

connorcook committed:
Updated package information

Reattributed the name to prolificinteractive, updates the version constraint on the MongoDB package to use a more modern version.

connorcook committed:
Update supported PHP versions

also make sure to use mongodb extension not mongo

ppatriotis commented on lib/Store/Store.php:
```diff
@@ -50,14 +57,14 @@ static function createConnectionURI($connectionDetails = array()) {
                 $connectionURI .= "&readPreference=${connectionDetails['readPreference']}";
             }
-        }
+        }*/
```
Commented code in PR ;)

ppatriotis commented on lib/Store/Store.php:
```diff
@@ -75,28 +82,36 @@ public function get($type, $key)
+
+        $cursor = $this->manager->executeQuery($namespace, $query);
+
+	    if (false === ($cursor = current($cursor->toArray()))) {
```
Looks like some formatting problem

ppatriotis commented on lib/Store/Store.php:
```diff
@@ -75,28 +82,36 @@ public function get($type, $key)
+        ];
+        $query = new Query($where, ['limit' => 1]);
+
+        $namespace = "{$this->dbName}.{$type}";
```
Don’t see namespace query or where used anywhere else

connorcook commented:
Haha, @ppatriotis I didn't think we'd see you around these parts again ;P

It's clear I'll need to tweak the work Jose did rather than straight up pulling it in. Gonna do it later, then, still got the Laravel Upgrade to finish first.

... and don't scare me like that, haha

Commit:
Fix styling issues, fix first test to see if it successfully compiles now, and remove commented out code

Commit:
Rewrite tests to reuse common tasks and use new MongoDB library

Commit:
Remove deprecated use of string in assert and replace with boolean arguments for compatability with 7.2

Commit:
Add support for PHP 5.6

Commit:
Add proper 5.5+ support to composer and Travis

connorcook commented:
@prolificchrisb I've fixed all the tests and addressed the feedback. It should be good to go, and it is now fully backwards compatible with PHP 5.5-5.6 as well as PHP7.0-7.2.

connorcook committed:
Correct supported versions of PHP in README

connorcook committed:
Correct supported version of PHP in composer package

prolificchrisb commented:
@connorcook I think we should create a separate repo, SimplesSAMLphp module, and Packagist package for mongodb and leave this repo targeted to the mongo extension.

connorcook committed:
Update composer schema for fork

Include new author

connorcook commented:
> @connorcook I think we should create a separate repo, SimplesSAMLphp module, and Packagist package for mongodb and leave this repo targeted to the mongo extension.

Sounds good.
